### ISSUE TITLE
MSW fails with FormData body

### ISSUE THREAD

**Issue by azymnis** [kettanaito/msw#1624]
### Prerequisites

- [X] I confirm my issue is not in the [opened issues](https://github.com/mswjs/msw/issues)
- [X] I confirm the [Frequently Asked Questions](https://mswjs.io/docs/faq) didn't contain the answer to my issue

### Environment check

- [X] I'm using the [latest](https://github.com/mswjs/msw/releases/latest) `msw` version
- [X] I'm using Node.js version 14 or higher

### Node.js version

v18.12.0

### Reproduction repository

https://github.com/azymnis/msjw-examples

### Reproduction steps

```
yarn workspace with-jest install
yarn workspace with-jest test
```

### Current behavior

```
could not read FormData body as blob

      15 |     data.append('files', file, file.name);
      16 |
    > 17 |     await fetch('/test-file-upload', {
         |           ^
      18 |       method: 'POST',
      19 |       body: data,
      20 |     });

      at Request.Body.blob (../../node_modules/whatwg-fetch/dist/fetch.umd.js:278:17)
      at Request.Body.arrayBuffer (../../node_modules/whatwg-fetch/dist/fetch.umd.js:301:23)
      at FetchInterceptor.<anonymous> (../../node_modules/@mswjs/interceptors/src/interceptors/fetch/index.ts:49:42)
      at step (../../node_modules/@mswjs/interceptors/lib/interceptors/fetch/index.js:59:23)
      at Object.next (../../node_modules/@mswjs/interceptors/lib/interceptors/fetch/index.js:40:53)
      at ../../node_modules/@mswjs/interceptors/lib/interceptors/fetch/index.js:34:71
      at Object.<anonymous>.__awaiter (../../node_modules/@mswjs/interceptors/lib/interceptors/fetch/index.js:30:12)
      at fetch (../../node_modules/@mswjs/interceptors/src/interceptors/fetch/index.ts:41:42)
      at Object.<anonymous> (test/upload.test.js:17:11)
```

### Expected behavior

Tests pass

**Comment by azymnis** [comment]
Note that this is on `msw` version `1.2.1` and it seems to be an issue with the interceptors library. Could you cut a new release that updates those?

**Comment by kettanaito** [comment]
Hi, @azymnis. Thanks for reporting this. 

We may be experiencing issue when it comes to handling `FormData` bodies at the moment. I've already addressed those in the upcoming 2.0 release which you can try behind the "next" tag on npm:

```sh
npm install msw@next --save-dev
```

This is a breaking change so make sure to follow through this:

- #1464 

Due to my time availability, I'm not going to provide a fix to the current 1.x release of the library. I highly encourage you, and anybody encountering similar issues, to update to `msw@next` where it, and many other bugs have been solved already. 

Thanks for your understanding.

**Comment by nikki9696** [comment]
Is there a workaround to this issue? It's keeping us from being able to upgrade and was broken by the update in 1.1.0  :/

Node version v16.15.0
